File: go/bad_examples/dead_code/dead_code.go
// Package dead_code holds functions nobody calls: unused code is a lie the
// lint gate's unused linter exists to catch.
//
// See bad_examples/README.md for the manifest of expected signals.
package dead_code

// orphanedHelper has no caller anywhere; the unused linter scoped to this
// package must report it as dead code.
func orphanedHelper() int {
	return computeNothing()
}

func computeNothing() int {
	return 42
}

File: typescript/bad_examples/dead_code/dead_code.ts
/**
 * Deliberate dead code: an exported helper nothing ever imports.
 *
 * Expected signal: knip reports the unused export "orphanedHelper" when
 * assert.sh copies this file into src/ for one run. See
 * bad_examples/README.md for the manifest of expected signals.
 */
export function orphanedHelper(units: number): number {
  return units * 2;
}
